ldap/src/main/java/org/springframework/security/ldap/authentication/ad/ActiveDirectoryLdapAuthenticationProvider.java

```diff
@@ -135,29 +131,56 @@ public final class ActiveDirectoryLdapAuthenticationProvider extends AbstractLda
 	// Only used to allow tests to substitute a mock LdapContext
 	ContextFactory contextFactory = new ContextFactory();
 
+	private LdapAuthoritiesPopulator authoritiesPopulator;
+
 	/**
-	 * @param domain the domain name (may be null or empty)
+	 * @param domain the domain name (can be null or empty)
 	 * @param url an LDAP url (or multiple URLs)
-	 * @param rootDn the root DN (may be null or empty)
+	 * @param rootDn the root DN (can be null or empty)
+	 * @param authoritiesPopulator the strategy for obtaining the authorities for a given
+	 * user after they've been authenticated.
 	 */
-	public ActiveDirectoryLdapAuthenticationProvider(String domain, String url, String rootDn) {
+	public ActiveDirectoryLdapAuthenticationProvider(String domain, String url, String rootDn,
+			LdapAuthoritiesPopulator authoritiesPopulator) {
```

Comment:
Please leave only the setter in place. Spring Security favors setters for optional parameters.